src/app/dialogs/import-experiment/import-experiment.component.ts

```diff
@@ -68,52 +69,51 @@ export class ImportExperimentDialog implements OnInit {
                 take(1)
             )
             .subscribe(experimentUpload => {
-                if (experimentUpload.importId) {
-                    // File upload complete. Poll for result.
-                    const importId: number = experimentUpload.importId;
-                    this.polling = "PENDING";
-                    this.uploading = false;
-                    interval(1000)
-                        .pipe(
-                            startWith(0),
-                            switchMap(() => this.backend.importExperimentPoll(importId)),
-                            filter(resp => resp.status != "PENDING"),
-                            take(1)
-                        )
-                        .subscribe(resp => {
-                            if (resp.status == "SUCCESS") {
-                                // close dialog after wait
-                                this.polling = "DONE"
-                                this.pollingResult = "Import successful. Forwarding to experiment...";
-                                this.waiting = true;
-                                setTimeout(() => this.dialogRef.close({ experimentId: resp.experimentId }), 2000);
-                            } else {
-                                if (resp.status == "FAILURE") {
-                                    console.error("Something went wrong. Check errors at backend.")
-                                } else {
-                                    console.error("Something went wrong. Backend returned wrong import status. Please check errors at backend.");
-                                }
-                                this.polling = "DONE";
-                                this.error = "Something went wrong.";
-                                this.reset();
-                            }
-                        })
-                } else {
-                    console.error("Something went wrong. Could not retrieve importId. Please check errors at backend.");
-                    this.error = "Something went wrong.";
-                    this.reset();
-                }
+                this.pollForImportResult(experimentUpload.importId)
             })
     }
 
+    pollForImportResult(importId: number | undefined) {
+        if (importId) {
```

Comment:
You can get rid of one level of nesting by reversing the if condition to fail fast

```typescript
if (importId == null) {
    // handle error
    return;
}
// rest of code
```